File: atcoder.jp/contests/abc135/tasks/abc135_b/main.cpp
#include <bits/stdc++.h>
using namespace std;

vector<int> input_vector(int n) {
  vector<int> ret(n);
  for (auto &&i : ret) cin >> i;
  return ret;
}

int main() {
  int N;
  cin >> N;
  auto P = input_vector(N);

  int cnt = 0;
  for (int i = 0; i < N; i++)
    if (i + 1 != P[i]) cnt++;

  bool ans = cnt == 0 || cnt == 2;

  cout << (ans ? "YES" : "NO") << endl;
}

File: atcoder.jp/contests/abc135/tasks/abc135_b/main.hs
import Util (count)

main :: IO ()
main = do
  n <- readLn
  p <- map read . words <$> getLine

  let cnt = count (== True) $ zipWith (/=) [1 .. n] p
      ans = cnt == 0 || cnt == 2

  putStrLn $ if ans then "YES" else "NO"
